OneCommunityGlobal/HGNRest
# Haoji fetch profile images from team page and sending email for selecting correct images

# Description
<img width="660" alt="image" src="https://github.com/OneCommunityGlobal/HGNRest/assets/99279909/c387b307-40cc-475b-8844-bcd7da85ba98">

## Related PRS (if any):
To test this backend PR you need to checkout the [#1531](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1531) frontend PR. 
The main changes are based on the previous PR: https://github.com/OneCommunityGlobal/HGNRest/pull/591.
…

## Main changes explained:
- Update package.json for Web Scraping tools
- Update src/controllers/userProfileController.js for allowing url as a valid img source
- Create a function updateProfilesPic in src/controllers/userProfileController.js for fetching the pictures from the sites
- Update src/models/userProfile.js to allow temporarily store the possible pictures
- Create an email sender for alerting the user if there are more than one possible imgs fetched by the cronjob
…

## How to test

1. check into current branch
2. do `npm install`, `npm run build` and `npm run start` to run this PR locally
3. log as owner user

4. ### **Please change the cronjob interval in src/cronjobs/userProfileJobs.js, the following change allows cronjob to run every minute. You could make the interval longer appropriately for testing. If you forget to change this, cronjob would only run everyday 12:00am**
<img width="660" alt="image" src="https://github.com/OneCommunityGlobal/HGNRest/assets/99279909/b6812e4c-3991-432e-ab96-2ed8d7a7b3f7">

5.  If you do not want to modify others' profiles during testing, you can add a search keyword in src/helpers/userHelper.js like the following. That could make sure cronjob only works on your own created users.
<img width="660" alt="image" src="https://github.com/OneCommunityGlobal/HGNRest/assets/99279909/6bf55b94-cf3c-44a9-97e2-56e7b27aa3a6">

6. See how to test in frontend
7. do `git restore .` to undo the changes

## Screenshots or videos of changes:

## Note:
See frontend
